File: Java_Programs/MatrixMul.java
public class MatrixMul {
    public static void main(String[] args) {
        int[][] A = {{1,2,3},{4,5,6}}; // 2x3
        int[][] B = {{7,8},{9,10},{11,12}}; // 3x2
        int[][] C = new int[2][2];
        for (int i = 0; i < 2; i++)
            for (int j = 0; j < 2; j++)
                for (int k = 0; k < 3; k++)
                    C[i][j] += A[i][k] * B[k][j];
        System.out.println("Result matrix:");
        for (int i = 0; i < 2; i++) {
            for (int j = 0; j < 2; j++) System.out.print(String.format("%4d", C[i][j]));
            System.out.println();
        }
    }
}

File: C_Programs/matrix_mul.c
#include <stdio.h>

int main(void) {
    int A[2][3] = {{1,2,3},{4,5,6}};
    int B[3][2] = {{7,8},{9,10},{11,12}};
    int C[2][2] = {{0}};

    for (int i = 0; i < 2; ++i)
        for (int j = 0; j < 2; ++j)
            for (int k = 0; k < 3; ++k)
                C[i][j] += A[i][k] * B[k][j];

    printf("Result matrix:\n");
    for (int i = 0; i < 2; ++i) {
        for (int j = 0; j < 2; ++j) printf("%4d", C[i][j]);
        printf("\n");
    }
    return 0;
}
